### PR TITLE
JU-599-revert custom company auth

### PR DESCRIPTION
**Pull request by mnycz-ch**
JU-599-revert custom company auth

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java
Comment (dhawes-ch) [inline src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:1]: Why do we need a custom implementation here vs https://github.com/companieshouse/web-security-java/blob/main/src/main/java/uk/gov/companieshouse/auth/filter/CompanyAuthFilter.java ?

Comment (mnycz-ch) [reply src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:1]: Hi Daniel, 
I think we need that to run custom checks in that service. As Lee mentioned to me that it company auth is required to all company at the moment and we should make some additional checks.

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * @param chain    the chain of filters for authorisation
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+
+        ValidatorResourceProvider resourceProvider = new ValidatorResourceProvider(apiClientService,
```
Comment (varshashastri) [inline src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:78]: Could you not inject these as autowired and mark this filter as a component?

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                formTemplateService);
+        Validator<HttpServletRequest> requiresAuth = new HttpRequestRequiredValidator(resourceProvider)
+                .setNext(new FormTemplateRequiredValidator(resourceProvider))
+                .setNext(new UserRequiredValidator(resourceProvider, categoryTemplateService));
+
+        if (requiresAuth.validate(httpServletRequest)) {
+            String companyNumber = resourceProvider.getCompanyNumber().orElse("");
+            Session chsSession = resourceProvider.getChsSession().orElse(null);
+
+            redirectForAuth(chsSession,
```
Comment (varshashastri) [inline src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:88]: Possibly this part is not tested properly the validators are not injected. Please check

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java (hunk 1 above)
Comment (vshastri-ch) [inline src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:78]: Could you not inject these?

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java (hunk 2 above)
Comment (vshastri-ch) [inline src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:88]: Here all cases are not tested properly as its not injected.

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java (hunk 1 above)
Comment (mnycz-ch) [reply src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:78]: I revert that piece of code back and do not make any change to that.

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java (hunk 2 above)
Comment (mnycz-ch) [reply src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:88]: I revert that piece of code back and do not make any change to that. This is not new piece of code.

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java
Comment (foman-ch) [reply src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:1]: Are you happy to resolve this conversation @dhawes-ch ? It does look like the custom implementation uses separate logic to that implemented in the web-security-java class.

**Review by foman-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java
Comment (foman-ch) [reply src/main/java/uk/gov/companieshouse/efs/web/security/CompanyAuthFilter.java:1]: Going to resolve this in Dan's absence, we can have a further discussion when he's back.

**Review by vshastri-ch — APPROVED** [review]

**Review by AlisonGriffiths — APPROVED** [review]

**Pull request merged**